src/akushon/action/model/action.cpp

```diff
@@ -108,6 +108,34 @@ const std::string & Action::get_next_action() const
 void Action::reset()
 {
   poses.clear();
+  for (auto & [id, spline] : joint_splines) {
+    delete spline;
+  }
+  joint_splines.clear();
+}
+
+void Action::enable_spline(bool enable)
+{
+  use_spline = enable;
+}
+
+bool Action::is_using_spline() const
+{
+  return use_spline;
+}
+
+void Action::generate_splines()
+{
+  for (auto pose : poses) {
+    for (auto & joint : pose.get_joints()) {
+      uint8_t joint_id = joint.get_id();
+
+      if (joint_splines.find(joint_id) != joint_splines.end()) {
+        joint_splines[joint_id] = new keisan::SmoothSpline();
```

Comment:
please use [`std::shared_ptr`](https://en.cppreference.com/w/cpp/memory/shared_ptr) instead, it's more safe. And you don't need to delete this later.